casts/splitNavigator/src/SplitNavigator/createSplitNavigator.tsx

```diff
@@ -7,86 +7,458 @@ import {
 };
+type SplitScreenOptions = SplitScreenTabBarIconOptions | Record<string, any>;
+
+function UnfoldedSplitNavigator({
+    navigation,
+    descriptors,
+    state,
+    mainRoute,
+    splitStyles,
+    tabRouteNamesMap,
+    loaded,
+    onTabPress,
+}: {
+    splitStyles: SplitStyles;
+    navigation: any;
+    descriptors: Record<
+        string,
+        Descriptor<
+            // eslint-disable-next-line @typescript-eslint/ban-types
+            Record<string, object | undefined>,
+            string,
+            SplitNavigationState<ParamListBase>,
+            SplitScreenOptions,
+            // eslint-disable-next-line @typescript-eslint/ban-types
+            {}
+        >
+    >;
+    state: SplitNavigationState<ParamListBase>;
+    mainRoute: NavigationRoute<ParamListBase, string>;
+    tabRouteNamesMap: Set<string>;
+    loaded: number[];
+    onTabPress: (key: string) => void;
+}) {
+    const tabBarIcons = React.useMemo(
+        () =>
+            state.routes.reduce<Record<string, SplitScreenTabBarIconOptions>>((acc, route) => {
+                if (!tabRouteNamesMap.has(route.name)) {
+                    return acc;
+                }
+
+                if (route.key === mainRoute.key) {
+                    return acc;
+                }
+
+                const descriptor = descriptors[route.key];
+                if (descriptor.options == null) {
+                    return acc;
+                }
+                if ('tabBarActiveIcon' in descriptor.options) {
+                    acc[route.key] = {
+                        tabBarActiveIcon: descriptor.options.tabBarActiveIcon,
+                        tabBarDisabledIcon: descriptor.options.tabBarDisabledIcon,
+                    };
+                    return acc;
+                }
+                if ('tabBarAnimatedIcon' in descriptor.options) {
+                    acc[route.key] = {
+                        tabBarAnimatedIcon: descriptor.options.tabBarAnimatedIcon,
+                    };
+                    return acc;
+                }
+
+                return acc;
+            }, {}),
+        // The rule is disabled to not include `descriptors` as a dep
+        // because descriptors going to be changed every render
+        // and will ruin the optimisation
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [tabRouteNamesMap, mainRoute.key, state.routes],
+    );
+
+    const doesSupportNative = Platform.OS !== 'web' && screensEnabled?.();
+    const tabBarHeight = useTabBarHeight();
+
+    return (
+        <NavigationHelpersContext.Provider value={navigation}>
+            <NestedInSplitContext.Provider value>
+                <SafeAreaProviderCompat>
+                    <View style={splitStyles.body}>
+                        <View style={splitStyles.main}>
+                            <SplitTabBarHeightContext.Provider value={tabBarHeight}>
+                                {descriptors[mainRoute.key].render()}
```

Comment:
This behaviour needs to be discussed with @eteslov.
I've imagined that the tab bar in the split view should still change the "left part" of the screen according to tabs content, but the navigation from that tab screen should then navigate to the "right part"